### PR TITLE
added Bamboo.LocalAdapter and updated versions

### PR DESCRIPTION
**Pull request by renesugar**
Added support for Bamboo.LocalAdapter for viewing emails in development.

Updated versions of dependencies.

### CODE REVIEW

**Comment by renesugar** [comment]
I ran "yarn check".

The changes in versions and .babelrc, webpack.config.js were to get rid of warnings and changes for Webpack 4.

https://babeljs.io/docs/en/env/

The Bamboo 1.0.0 uses "Bamboo.SentEmailViewerPlug" instead of "Bamboo.EmailPreviewPlug".

File: apps/subs_web/frontend/webpack.config.js [hunk 1]
```diff
@@ -47,6 +49,19 @@ module.exports = {
     path: path.resolve(__dirname, '../priv/static'),
     filename: '[name].bundle.js',
   },
+  /*
+  optimization: {
```
Comment (joaquimadraz) [inline apps/subs_web/frontend/webpack.config.js]: Can we remove this commented code?

File: apps/subs_web/frontend/webpack.config.js [hunk 2]
```diff
@@ -3,6 +3,8 @@ const webpack = require('webpack')
 const ExtractTextPlugin = require('extract-text-webpack-plugin')
 
 module.exports = {
+  mode: 'development',
```
Comment (joaquimadraz) [inline apps/subs_web/frontend/webpack.config.js]: Given that we don't have a production config for webpack, is this going to affect the production build?

File: apps/subs_web/frontend/webpack.config.js [hunk 3]
```diff
@@ -74,7 +89,6 @@ module.exports = {
   },
   plugins: [
     new ExtractTextPlugin({ filename: 'app.bundle.css', allChunks: true }),
-    new webpack.optimize.CommonsChunkPlugin({ name: 'vendor', filename: 'vendor.bundle.js' }),
```
Comment (joaquimadraz) [inline apps/subs_web/frontend/webpack.config.js]: What's the reasoning behind removing the vendor bundle?

**Comment by renesugar** [comment]
I am building the project on Mac OS Version 10.13.5.

I added another commit to back out the version changes.

I've attached the output of "yarn install" when I back out the version changes.
[yarn_install.txt](https://github.com/joaquimadraz/opensubs.io/files/2153233/yarn_install.txt)

File: apps/subs_web/frontend/webpack.config.js (hunk 3 above)
Comment (renesugar) [reply apps/subs_web/frontend/webpack.config.js]: I removed the changes for Webpack 4.

https://webpack.js.org/plugins/commons-chunk-plugin/

"The CommonsChunkPlugin has been removed in webpack v4 legato. To learn how chunks are treated in the latest version, check out the SplitChunksPlugin."

File: apps/subs_web/frontend/webpack.config.js (hunk 1 above)
Comment (renesugar) [reply apps/subs_web/frontend/webpack.config.js]: I added another commit to the pull request to remove the front end changes.

File: apps/subs_web/frontend/webpack.config.js (hunk 2 above)
Comment (renesugar) [reply apps/subs_web/frontend/webpack.config.js]: The mode changes the default plugins.

https://webpack.js.org/concepts/mode/

After moving to Webpack 4, a better place to set the mode for dev vs production is in package.json:

"scripts": {
  "dev": "webpack --mode development",
  "build": "webpack --mode production"
}

I removed the changes for Webpack 4.

**Comment by joaquimadraz** [comment]
Thank you! 🙌 Sorry for the delay @renesugar.

**Pull request merged**